### ISSUE TITLE
Why is datastore initialized differently than the other libraries?

### ISSUE THREAD

**Issue by ryanseys** [GoogleCloudPlatform/gcloud-node#593]
See [here](https://github.com/GoogleCloudPlatform/gcloud-node/blob/8c3719ff57f0a03165cc18251e9b57494b94bd0f/lib/index.js#L118) for what I'm talking about. Seems that once `gcloud` is initialized, datastore cannot be initialized with different options. I think this is because we delegated it to the `dataset` object instead but that just is kinda messy. Would it be worth it to change this to be consistent with the other APIs?

Current way:
```js
var gcloud = require('gcloud')(options);
var datastore = gcloud.datastore;
var dataset = datastore.dataset(newOptions);
```

Proposed way:
``` js
var gcloud = require('gcloud')(options);
var datastore = gcloud.datastore(newOptions);
var dataset = datastore.dataset();
```

ADDITIONAL CRAZY IDEA: What if we just eliminated the `dataset` altogether? Something like:

``` js
var gcloud = require('gcloud')(options);
var datastore = gcloud.datastore(newOptions);
var key = datastore.key('Company');
datastore.save(key, { data: { hello: 'world' } }, callback);
```

Anything preventing us from doing this?

**Comment by stephenplusplus** [comment]
I would really like to do the last one. @rakyll said to anticipate more things added to Datastore, which was why we couldn't do it.

**Comment by jgeewax** [comment]
I'm down for:

```javascript
var gcloud = require('gcloud')(options);
var datastore = gcloud.datastore(newOptions);
var dataset = datastore.dataset('name of the dataset' /* optional */);
var key = datastore.key('Company');
datastore.save(key, { data: { hello: 'world' } }, callback);
```

The reason we can't kill data set is that we are in the process of adding the ability to have multiple datasets (we'll probably call them databases) per project, so `datastore.save()` is a misleading concept...

But I still think we need to allow you to configure stuff at the datastore level... ? right?

**Comment by ryanseys** [comment]
Oh, as far as I understand right now we have no concept of a name of a
dataset, right? In that case I like my first new proposal where it will
just take no options for the time being until this multi-dataset
functionality rolls around. This would be a breaking change.

**Comment by jgeewax** [comment]
`datastore.dataset()` should always still work, and default to the ID of the project. In the future, we need a way of passing in a name of a dataset (`datastore.dataset('name')`).

Don't think this should be a breaking change, right?

**Comment by ryanseys** [comment]
It would be breaking because gcloud.datastore will now be a function that
needs to be called with options. datastore.dataset will no longer accept
these options. And it doesn't make sense to have it in both places.

**Comment by jgeewax** [comment]
Gotcha - we couldn't make it work with and without the parens? (`gcloud.datastore` == `gcloud.datastore()`) ?

**Comment by ryanseys** [comment]
Maybe but if we're in the business of breaking things with the moving of
options from dataset to datastore then we should just break it. Also this
is much easier to do pre-1.0

**Comment by stephenplusplus** [comment]
Just want to make sure this is correct:

```js
// uses dataset-id-1:
var gcloud = require('gcloud')({ projectId: 'dataset-id-1' });
gcloud.datastore.dataset();

// uses dataset-id-2:
var gcloud = require('gcloud')({ projectId: 'dataset-id-1' });
var datastore = gcloud.datastore({ projectId: 'dataset-id-2' });
datastore.dataset();

// uses dataset-id-3:
var gcloud = require('gcloud')({ projectId: 'dataset-id-1' });
var datastore = gcloud.datastore({ projectId: 'dataset-id-2' });
datastore.dataset('dataset-id-3'); // the projectId given above is useless.
```

**Comment by stephenplusplus** [comment]
@callmehiphop does this change how you would approach your PR (#845) at all?

**Comment by callmehiphop** [comment]
I don't believe so, I think it would simplify the api object creation though.

**Comment by stephenplusplus** [comment]
This is going to be not-a-thing when datastore v1beta3 is released (see #897). The `dataset` concept is gone and there is only a "datastore" per project.
